test/Android/PrepareEmulator.ps1

```diff
@@ -0,0 +1,86 @@
+[CmdletBinding()]
+Param(
+    [switch]$killRunningEmulator
+)
+
+$xdeProcess = Get-Process | where {$_ -like "*XDE*"}
+if ($xdeProcess.count -gt 0)
+{
+    if ($killRunningEmulator)
+    {
+        <# Kill emulator if already started, otherwise the hanging issue is more likely to repro. #>
+        $xdeProcess | Stop-Process
+    }
+    else 
+    {
+        Write-Output "Emulator already started"
+        exit 0
+    }
+}
+
+$emulatorPath = ${Env:\ProgramFiles(x86)} + '/Microsoft Emulator Manager/1.0'
+
+if (!(Test-Path $emulatorPath))
+{
+    Write-Error "Emulator manager is not installed at $emulatorPath"
+    exit 1
+}
+
+pushd $emulatorPath
+
+$deviceList = ./emulatorcmd list /sku:Android /type:device
+
+<# try to use Lollipop emulator if there is one #>
+$device = $deviceList | where {$_ -match ".*(Lollipop).*" } | Select-Object -First 1
+if (!$device) 
+{
+    $device = $deviceList | where {$_ -match ".*(Marshmallow|KitKat).*" } | Select-Object -First 1
+
+    if (!$device)
+    {
+        Write-Error 'Can not find an valid emulator device'
+        popd
+        exit 1    
+    }
+}
+
+
+<# launch emulator #>
+Write-Output "Starting device: $device"
+
+$deviceId = $device.Substring(0, $device.IndexOf(' '))
+start-process powershell -argumentlist "emulatorcmd.exe launch /sku:android /id:$deviceId"
+popd
+
+<# wait for emulator to be ready #>
+$adbPath = ${Env:\ProgramFiles(x86)} + '/Android/android-sdk/platform-tools'
```

Comment:
Most likely not worth it (given this will run on a test machine we control), but just in case you weren't aware: there are registry keys with the path to the Android SDK.

Software\Android SDK Tools
Software\Android Studio